classes/Tools.php

```diff
@@ -1024,20 +1024,36 @@ public static function clearXMLCache()
     }
 
     /**
-    * Display an error according to an error code
+    * Depending on _PS_MODE_DEV_ throws an exception or returns a error message
     *
-    * @param string $string Error message
-    * @param bool $htmlentities By default at true for parsing error message with htmlentities
+    * @param string|null $errorMessage Error message (defaults to "Fatal error")
+    * @param bool $htmlentities DEPRECATED since 1.7.4.0
+    * @param Context|null $context DEPRECATED since 1.7.4.0
+    *
+    * @return string
+    *
+    * @throws PrestaShopException If _PS_MODE_DEV_ is enabled
     */
-    public static function displayError($string = 'Fatal error', $htmlentities = true, Context $context = null)
+    public static function displayError($errorMessage = null, $htmlentities = null, Context $context = null)
```

Comment:
you can't do that, this will break the API.

What you can do is creating a new function and then replace every call of  `displayError` buy a new one (the best place of this new function should probably be in Adapter/Tools class

Reply:
@mickaelandrieu As far as I can see the only API break is that if the method is called with the default(!) parameter $string='Fatal error' because it won't be translated anymore. Or is there something else? I can't see another change on the API.. Is it really necessary to translate 'Fatal error' if it is explicitly provided?

Reply:
awwww just saw that "htmlentities" argument was not used anymore o_O